src/checkout-buttons/strategies/googlepay/googlepay-button-strategy.ts

```diff
@@ -83,7 +83,13 @@ export default class GooglePayButtonStrategy extends CheckoutButtonStrategy {
 
         return this._googlePayPaymentProcessor.displayWallet()
             .then(paymentData => this._googlePayPaymentProcessor.handleSuccess(paymentData)
-                .then(() => this._googlePayPaymentProcessor.updateShippingAddress(paymentData.shippingAddress)))
+            .then(() => {
+                if (paymentData.shippingAddress) {
+                    this._googlePayPaymentProcessor.updateShippingAddress(paymentData.shippingAddress);
+                }
+
+                return Promise.resolve();
```

Comment:
You actually don't need this line. The executor function of the promise only needs to return a `Promise<T>` if you need to perform an additional asynchronous operation. In this case, you don't therefore it's better if you omit this line.